MapleServer2/PacketHandlers/Game/Helpers/ItemUseHelper.cs

```diff
@@ -0,0 +1,150 @@
+
+                    foreach (ItemContent item in content)
+                    {
+                        if (success)
+                        {
+                            if (ItemMetadataStorage.GetRecommendJobs(item.Id).Contains(session.Player.JobGroupId)) // maybe this need to a random call instead to get the next item on Content
+                            {
+                                GiveItem(session, item);
+                                break;
+                            }
+                        }
+                        else
+                        {
+                            if (ItemMetadataStorage.GetRecommendJobs(item.Id).Contains(session.Player.JobGroupId))
+                            {
+                                content.Remove(item);
+                                break;
+                            }
+                        }
+                    }
+
+                    if (!success)
+                    {
+                        int rand = rng.Next(0, content.Count);
+                        GiveItem(session, content[rand]);
+                    }
+                }
+            }
+            else
+            {
+                foreach (ItemContent item in content)
+                {
+                    if (ItemMetadataStorage.GetRecommendJobs(item.Id).Contains(session.Player.JobGroupId) || ItemMetadataStorage.GetRecommendJobs(item.Id).Contains(0))
+                    {
+                        GiveItem(session, item);
+                    }
+                }
+            }
+        }
+
+        public static void GiveItem(GameSession session, ItemContent content)
+        {
+            Random rng = new Random();
+
+            // Currency
+            if (content.Id.ToString().StartsWith("9"))
+            {
+                switch (content.Id)
+                {
+                    case 90000001: // Meso
+                        session.Player.Wallet.Meso.Modify(rng.Next(content.MinAmount, content.MaxAmount));
+                        break;
+                    case 90000004: // Meret
+                    case 90000011: // Meret
+                    case 90000015: // Meret
+                    case 90000016: // Meret
+                        session.Player.Wallet.Meret.Modify(rng.Next(content.MinAmount, content.MaxAmount));
+                        break;
+                }
+            }
+            // Items
+            else
+            {
+                Item item = new Item(content.Id)
+                {
+                    Amount = rng.Next(content.MinAmount, content.MaxAmount),
+                    Rarity = content.Rarity,
+                    Enchants = content.EnchantLevel,
+                };
+                InventoryController.Add(session, item, true);
+
+                if ((item.RecommendJobs.Contains(70) || item.RecommendJobs.Contains(80)) && item.ItemSlot == ItemSlot.OH)
+                {
+                    item = new Item(content.Id)
```

Comment:
Why are we adding the same item twice?

Reply:
Assassins and thieves should receive two weapons, simply using Item Amount does not work because equips can't have more than 1 amount.